Mathlib/Condensed/Discrete/LocallyConstant.lean

```diff
@@ -0,0 +1,442 @@
+  hom := unit P hs
+  inv := { app := fun X f ↦ f.toFun PUnit.unit }
+
+/--
+`CompHausLike.LocallyConstant.functor` is left adjoint to the forgetful functor.
+-/
+@[simps]
+noncomputable def adjunction [HasExplicitFiniteCoproducts.{u} P] :
+    functor.{u, w} P hs ⊣ (sheafSections _ _).obj ⟨CompHausLike.of P PUnit.{u+1}⟩ where
+  unit := unit P hs
+  counit := counit P hs
+  left_triangle_components := by
+    intro X
+    simp only [Functor.comp_obj, Functor.id_obj, NatTrans.comp_app, Functor.flip_obj_obj,
+      sheafToPresheaf_obj, functor_obj_val, functorToPresheaves_obj_obj, coe_of, whiskerRight_app,
+      Functor.associator_hom_app, whiskerLeft_app, Category.id_comp, NatTrans.id_app']
+    apply Sheaf.hom_ext
+    rw [Sheaf.instCategorySheaf_comp_val, Sheaf.instCategorySheaf_id_val]
+    exact locallyConstantAdjunction_left_triangle P hs X
+  right_triangle_components := by
+    intro X
+    ext (x : X.val.obj _)
+    simp only [Functor.comp_obj, Functor.id_obj, Functor.flip_obj_obj, sheafToPresheaf_obj,
+      FunctorToTypes.comp, whiskerLeft_app, unit_app, coe_of, Functor.associator_inv_app,
+      functor_obj_val, functorToPresheaves_obj_obj, types_id_apply, whiskerRight_app,
+      Functor.flip_obj_map, sheafToPresheaf_map, counit_app_val, counitApp_app, NatTrans.id_app']
+    have := CompHausLike.preregular hs
+    let _ : PreservesFiniteProducts
+        ((sheafToPresheaf (coherentTopology (CompHausLike P)) (Type (max u w))).obj X) :=
+      inferInstanceAs (PreservesFiniteProducts (Sheaf.val _))
+    apply presheaf_ext ((unit P hs).app _ x)
+    intro a
+    erw [incl_of_counitAppApp]
+    simp only [sheafToPresheaf_obj, unit_app, coe_of, counitAppAppImage,
+      LocallyConstant.coe_const]
+    have := map_eq_image _ a ⟨PUnit.unit, by
+      simp [mem_iff_eq_image (a := a), ← map_preimage_eq_image]⟩
+    erw [← this]
+    simp only [coe_of, unit_app, LocallyConstant.coe_const, Function.const_apply]
+    congr
+
+instance [HasExplicitFiniteCoproducts.{u} P] : IsIso (adjunction P hs).unit :=
+  inferInstanceAs (IsIso (unitIso P hs).hom)
+
+end Adjunction
+
+end CompHausLike.LocallyConstant
+
+section Condensed
+
+open Condensed CompHausLike
+
+/-- The functor from sets to condensed sets given by locally constant maps into the set. -/
+abbrev CondensedSet.LocallyConstant.functor : Type (u+1) ⥤ CondensedSet.{u} :=
+  CompHausLike.LocallyConstant.functor.{u, u+1} (P := fun _ ↦ True)
+    (hs := fun _ _ _ ↦ ((CompHaus.effectiveEpi_tfae _).out 0 2).mp)
+
+/--
+`CondensedSet.LocallyConstant.functor` is isomorphic to `Condensed.discrete`
+(by uniqueness of adjoints).
+-/
+noncomputable def CondensedSet.LocallyConstant.iso :
+    CondensedSet.LocallyConstant.functor ≅ discrete (Type (u+1)) :=
+  (LocallyConstant.adjunction _ _).leftAdjointUniq (discreteUnderlyingAdj _)
+
+/-- `CondensedSet.LocallyConstant.functor` is fully faithful. -/
+noncomputable def fullyFaithfulCondensedSetLocallyConstantFunctor :
```

Comment:
What do you think of a name like this? Some of the other names could maybe also be shortened-modulo-namespacing in a similar way.
```suggestion
noncomputable def CondensedSet.LocallyConstant.functor_fullyFaithful :
```

Reply:
Yes that makes sense, but I guess it would be `CondensedSet.LocallyConstant.functorFullyFaithful` because `FullyFaithful` isn't a prop. I'll make these changes tomorrow.